src/components/stepforms/widgets/DateComponents/RelativeDateForm.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div class="widget-relative-date-form">
+    <InputNumberWidget class="widget-relative-date-form__quantity" v-model="quantity" :min="1" />
+    <AutocompleteWidget
+      class="widget-relative-date-form__duration"
+      v-model="duration"
+      :options="durations"
+      trackBy="value"
+      label="label"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+import AutocompleteWidget from '@/components/stepforms/widgets/Autocomplete.vue';
+import InputNumberWidget from '@/components/stepforms/widgets/InputNumber.vue';
+import { DEFAULT_DURATIONS, DurationOption, RelativeDate } from '@/lib/dates';
+/**
+ * This component return a relative date composed from a number and a "duration" variable
+ */
+@Component({
+  name: 'relative-date-form',
+  components: {
+    InputNumberWidget,
+    AutocompleteWidget,
+  },
+})
+export default class RelativeDateForm extends Vue {
+  @Prop({ default: () => ({ quantity: 1, duration: DEFAULT_DURATIONS[0].value }) })
+  value!: RelativeDate;
+
+  @Prop({ default: false })
+  isNegative!: boolean; // define if the quantity should be negative
+
+  get quantity(): number {
+    return Math.abs(this.value.quantity);
+  }
+
+  set quantity(quantity: number) {
+    const abs = this.isNegative ? -1 : 1;
+    this.$emit('input', { ...this.value, quantity: quantity * abs });
+  }
+
+  get durations(): DurationOption[] {
+    /* When quantity is negative we add "ago" to durations labels
+     in order to specify user that relative date refers to past */
+    if (this.isNegative) {
+      return [...DEFAULT_DURATIONS].map(opt => ({
+        ...opt,
+        label: opt.label + ' ago',
+      }));
```

Comment:
Je me demande dans quel cas on opérera pas sur une date relative "négative" / dans le passé par rapport au présent.
Est ce qu'on a un exemple dans les mockups ?

Reply:
Je l'ai fais suite aux créa et à la PAT que t'as mis en place à vrai dire vu qu'il y avait des valeurs négatives dedans :thinking: . Je pense que ça vaux le coup de le garder pour le moment, je sais pas si on considère qui du from ou du to est dans le présent ou dans le passé mais du coup dans l'un des deux cas il faudra potentiellement avoir une logique comme celle là donc c'est pas perdu

Reply:
Je l'ai retiré dans ce commit https://github.com/ToucanToco/weaverbird/pull/988/commits/621876aeb74548d6979a9b4f0b9dc7c9081f0fda